ChatConnection.go

```diff
@@ -9,7 +9,7 @@ import (
 
 // ChatConnection : A wrapper for net.TCPConn to provide chat convencience methods
 type ChatConnection struct {
-	*net.TCPConn
+	net.Conn
```

Comment:
on line 28 in main.go, the compiler was complaining about using `net.Conn` instead of `*net.TCPConn` so it was just one of those necessity changes. I need to learn more about why